### DISCUSSION TITLE
3.0.0-dev.4

### DISCUSSION THREAD

**Discussion by tomschr** [python-semver/python-semver#387] [Announcements]
Related pr: https://github.com/python-semver/python-semver/issues/385

* Released
  2022-01-20

* Maintainer
  Tom Schraitle

**PyPI**: https://pypi.org/project/semver/3.0.0.dev4/
**Documentation:** https://python-semver.rtfd.io/en/3.0.0-dev.4/
**Full Changelog:** https://github.com/python-semver/python-semver/compare/3.0.0-dev.3...3.0.0-dev.4

# Bug Fixes
* #374: Correct Towncrier’s config entries in the pyproject.toml file. The old entries `[[tool.towncrier.type]]` are deprecated and need to be replaced by `[tool.towncrier.fragment.<TYPE>]`.

# Deprecations
* #372: Deprecate support for Python 3.6
  Python 3.6 reached its end of life and isn’t supported anymore. At the time of writing (Dec 2022), the lowest version is 3.7.

   Although the poll didn’t cast many votes, the majority agree to remove support for Python 3.6.

# Improved Documentation
* #335: Add new section “Converting versions between PyPI and semver” the limitations and possible use cases to convert from one into the other versioning scheme.
* #340: Describe how to get version from a file
* #343: Describe combining Pydantic with semver in the “Advanced topic” section.
* #350: Restructure usage section. Create subdirectory “usage/” and splitted all section into different files.
* #351: Introduce new topics for:
   * “Migration to semver3”
   * “Advanced topics”

# Features
* PR #359: Add optional parameter `optional_minor_and_patch` in `Version.parse()` to allow optional minor and patch parts.
* PR #362: Make `Version.match()` accept a bare version string as match expression, defaulting to equality testing.
* #364: Enhance pyproject.toml to make it possible to use the pyproject-build command from the build module. For more information, see Building semver.
* #365: Improve pyproject.toml.
   * Use setuptools, add metadata. Taken approach from A Practical Guide to Setuptools and Pyproject.toml.
   * Doc: Describe building of semver
   * Remove `.travis.yml` in `MANIFEST.in` (not needed anymore)
   * Distinguish between Python 3.6 and others in tox.ini
   * Add skip_missing_interpreters option for tox.ini
   * GH Action: Upgrade setuptools and setuptools-scm and test against 3.11.0-rc.2

# Trivial/Internal Changes
* #378: Fix some typos in Towncrier configuration

# Contributors
(in alphabetical order)

* @b0uh (new contributor)
* @calebstewart
* @jmi2k
* @Nagidal  (new contributor)
* @OidaTiftla 

Thanks to all! :heart: 

<hr /><em>This discussion was created from the release <a href='https://github.com/python-semver/python-semver/releases/tag/3.0.0-dev.4'>3.0.0-dev.4</a>.</em>

**Comment by skapin** [comment]
Hello, when do you plan to land version 3 ?

**Reply by tomschr** [reply]
Hi @skapin,
that's a good question. I was working on the PR #368 which I would like to integrate in the final version. Perhaps for #365 too.

When at least the first is integrated, I would like to create the final 3.0.0 release. Hope that helps. :slightly_smiling_face:
